xl-v1-namespace.go

```diff
@@ -0,0 +1,61 @@
+/*
+ * Minio Cloud Storage, (C) 2016 Minio, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package main
+
+import "sync"
+
+type nameSpaceParam struct {
+	volume string
+	path   string
+}
+
+type nameSpaceLock struct {
+	rwMutex *sync.RWMutex
+	count   uint
+}
+
+func (nsLock nameSpaceLock) InUse() bool {
+	return nsLock.count != 0
+}
+
+func (nsLock nameSpaceLock) Lock() {
```

Comment:
> needs to be a pointer receiver here:
> func (nsLock *nameSpaceLock) Lock() {
> 
> as nsLock.count++ will be lost

When used in this manner it is better to return nameSpaceLock itself as return value, so it can be copied as new values. 

But since it's not safe, it is better to do was krishna suggested, since that is the intention here.  

```
package main

import (
	"fmt"
)

type test struct {
	count uint
}

func (t test) Increment() test {
	t.count++
	return t
}

func (t test) String() string {
	return fmt.Sprintf("Count %d", t.count) 
}

func main() {
	t := test{}
	fmt.Println(t.Increment().Increment())
}
```

```
Count 2
```